src/api/DocumentApi.ts

```diff
@@ -53,16 +57,14 @@ export const useDocumentSearch = () => {
   const [loading, setLoading] = React.useState<boolean>(false);
 
   useEffect(() => {
-    const search = async () => {
+    (async () => {
       if (documentSearch && documentSearch.length > 2) {
         setLoading(true)
-        const res = await searchDocuments(documentSearch)
-        setDocumentSearchResult(res.content)
+        setDocumentSearchResult((await searchDocuments(documentSearch)).content)
         setLoading(false)
       }
-    }
-    search()
+    })()
   }, [documentSearch])
 
-  return [documentSearchResult, setDocumentSearch, loading] as [Document[], Dispatch<SetStateAction<string>>, boolean]
+  return [documentSearchResult, setDocumentSearch, loading, setDocumentSearchResult] as [Document[], Dispatch<SetStateAction<string>>, boolean, Function]
```

Comment:
Ikke sett internresultatet i denne hooken fra utsiden, sett heller søkeordet, og om søkeord er null så tøm resultat i denne hooken

Reply:
Når dokumenten slettes, må jeg oppdatere select. Jeg brukte setDocument men det funket ikke.

Reply:
Nei, for du må endre denne hooken til å slette resultat om søkeordet er tomt, i hooken her så skjer det ingenting om det ikke er søketekst